src/unitxt/operators.py

```diff
@@ -207,7 +215,12 @@ def process(
 
     def get_mapped_value(self, instance, key, mapper, val):
         val_as_str = str(val)  # make sure the value is a string
-        if val_as_str in mapper:
+        if (
+            val_as_str in mapper
+        ):  # carefully check what to return, to avoid external modification to the mapped value (e.g. if it is a list)
+            do_cast, cast_what = self._cast_returned_value[key][val_as_str]
+            if do_cast:
+                return ast.literal_eval(cast_what)
```

Comment:
what of the purpose of this stringfication and then de-serialization?

Reply:
Only of values (to return from the mapping) that are not strings. 
An easy way to generate a new (again: not string) object, to return from the mapping.
On verification time, all the non-string values to return are identified and marked.
Then, on run time, a tiny overhead of looking up that mark is incurred, and then **only the non-strings** go through  `ast.literal_eval`  .  The original string representation of the value is maintained intact at the operator.

Reply:
So is this a way to achieve deep copy for simple python strucres?

Reply:
@dafna @elronbandel - The original deep copy is a reasonable solution for string.  Deepcopy does not  copy immutable objects.

(env2) yoavkatz@Yoavs-MacBook-Pro unitxt % cat a.py                              
import copy

original_string = "hello"
copied_string = copy.deepcopy(original_string)

print(id(original_string))  # Output: some memory address
print(id(copied_string))  # Output: the same memory address 
(env2) yoavkatz@Yoavs-MacBook-Pro unitxt % python a.py
4343111536
4343111536

Reply:
Also we should change:

Attributes:
mappers (Dict[str, Dict[str, str]]): The mappers to use for mapping instance values.
Keys are the names of the fields to be mapped, and values are dictionaries
that define the mapping from old values to new values.

to 

Attributes:
mappers (Dict[str, Dict[str, **Any**]]): The mappers to use for mapping instance values.
Keys are the names of the fields to be mapped, and values are dictionaries
that define the mapping from old values to new values.